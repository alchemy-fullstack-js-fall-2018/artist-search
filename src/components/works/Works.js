import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';


const Work = ({ title, name }) => {
  return (
    <Fragment>
      <h3>{title}</h3>
      <Link to={ROUTES.SONG.linkTo(name, title)}>Get Lyrics</Link>
    </Fragment>
  );
};

export default function Works({ works, name }) {
  return (
    <Fragment>
      {works.map(work => <Work key={work.id} title={work.title} name={name} />)}
    </Fragment>
  );
}
