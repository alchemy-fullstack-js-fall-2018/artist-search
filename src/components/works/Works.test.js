import React from 'react';
import Works from './Works';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';


describe('Works', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Works works={[]} />);
  });

  it('renders Works', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays song title', () => {
    const wrapperWithRouter = shallow(
      <MemoryRouter>
        <Works works={[{ id: '12',  title: 'a song', name: 'meh' }]} />
      </MemoryRouter>
    );
    expect(wrapperWithRouter.html()).toContain('a song');
  });

});
