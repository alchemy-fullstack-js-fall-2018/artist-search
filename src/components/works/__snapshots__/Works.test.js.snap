// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Works renders Works 1`] = `<Fragment />`;
